table/table.go

```diff
@@ -101,108 +92,370 @@ func DefaultKeyMap() KeyMap {
 func WithFocused(f bool) Option {
 	return func(m *Model) {
 		m.focus = f
 	}
 }
 
-// WithStyles sets the table styles.
+// WithStyles sets the table styles. This function is used as an [Option] in
+// when creating a table with [New].
 func WithStyles(s Styles) Option {
 	return func(m *Model) {
-		m.styles = s
+		m.SetStyles(s)
+	}
+}
+
+// WithStyleFunc sets the table [table.StyleFunc] for conditional styling. This
+// function is used as an [Option] in when creating a table with [New].
+func WithStyleFunc(s table.StyleFunc) Option {
+	return func(m *Model) {
+		m.useStyleFunc = true
+		m.table.StyleFunc(s)
 	}
 }
 
-// WithKeyMap sets the key map.
+// WithKeyMap sets the [KeyMap]. This function is used as an [Option] in when
+// creating a table with [New].
 func WithKeyMap(km KeyMap) Option {
 	return func(m *Model) {
 		m.KeyMap = km
 	}
 }
 
-// Update is the Bubble Tea update loop.
+// Setters
+
+// SetHeaders sets the table headers.
+func (m *Model) SetHeaders(headers ...string) *Model {
+	m.headers = headers
+	m.table.Headers(headers...)
+	return m
+}
+
+// SetRows sets the table rows.
+func (m *Model) SetRows(rows ...[]string) *Model {
+	m.rows = rows
+	m.table.Rows(rows...)
+	return m
+}
+
+// SetCursor sets the cursor position in the table.
+func (m *Model) SetCursor(n int) *Model {
+	m.cursor = clamp(n, 0, len(m.rows)-1)
+	return m
+}
+
+// SetHeight sets the width of the table. The given height will be the total
+// table height including borders, margins, and padding.
+func (m *Model) SetHeight(h int) *Model {
+	m.table.Height(h)
+	return m
+}
+
+// SetWidth sets the width of the table. The given width will be the total
+// table width including borders, margins, and padding.
+func (m *Model) SetWidth(w int) *Model {
+	m.table.Width(w)
+	return m
+}
+
+// SetFocused sets the focus state of the table.
+func (m *Model) SetFocused(f bool) *Model {
+	m.focus = f
+	return m
+}
+
+// SetYOffset sets the YOffset position in the table.
+func (m *Model) SetYOffset(n int) *Model {
+	m.yOffset = clamp(n, 0, len(m.rows)-1)
+	m.table.YOffset(m.yOffset)
+	return m
+}
+
+// SetStyles sets the table styles, only applying non-empty [Styles]. Note: using
+// [Model.SetStyleFunc] will override styles set in this function.
+func (m *Model) SetStyles(s Styles) *Model {
+	if !reflect.DeepEqual(s.Selected, lipgloss.Style{}) {
+		m.styles.Selected = s.Selected
+	}
+	if !reflect.DeepEqual(s.Header, lipgloss.Style{}) {
+		m.styles.Header = s.Header
+	}
+	if !reflect.DeepEqual(s.Cell, lipgloss.Style{}) {
+		m.styles.Cell = s.Cell
+	}
+	return m
+}
+
+// OverwriteStyles sets the table styles, overwriting all existing styles. Note:
+// using [Model.SetStyleFunc] will override styles set in this function.
+func (m *Model) OverwriteStyles(s Styles) *Model {
+	m.styles = s
+	return m
+}
+
+// OverwriteStylesFromLipgloss sets the [Model]'s style attributes from an
+// existing [lipgloss.Table].
+func (m *Model) OverwriteStylesFromLipgloss(t *table.Table) {
```

Comment:
Not sure if we could improve the name of this function...

Reply:
We could probably make it `StyleFromLipgloss` ?